bindings/generator/generate.py

```diff
@@ -505,42 +505,44 @@ class ParsingPlaceholder:
     )
 
 
-def generate_client(api_specs: ApiSpecs) -> list[str]:
-    template = env.get_template("client_plugin_definitions.d.ts.j2")
-    output = (BASEDIR / "../../client/src/plugins/libparsec/definitions.d.ts").resolve()
-    print(f"Generating {output}")
-    # Don't use `write_text` given it outputs \r\n for newlines on Windows
-    output.write_bytes(template.render(api=api_specs).encode("utf8"))
-    return []
+def gen_generate_fn(
+    template: str, dest: Path, modified_crate: str | None = None
+) -> Callable[[ApiSpecs], str | None]:
+    dest = dest.resolve()
 
+    def _generate(api_specs: ApiSpecs) -> str | None:
+        _template = env.get_template(template)
+        print(f"Generating {dest}")
+        # Don't use `write_text` given it outputs \r\n for newlines on Windows
+        dest.write_bytes(_template.render(api=api_specs).encode("utf8"))
+        return modified_crate
 
-def generate_electron(api_specs: ApiSpecs) -> list[str]:
-    template = env.get_template("binding_electron_meths.rs.j2")
-    output = (BASEDIR / "../electron/src/meths.rs").resolve()
-    print(f"Generating {output}")
-    output.write_bytes(template.render(api=api_specs).encode("utf8"))
+    return _generate
```

Comment:
IMO this is overcomplex and makes the code less easy to understand. Always prioritize readability first.

If the goal is to avoid duplication, why not simply abstract the common code into a function receiving the necessary parameters?

```python
generate_target(api_specs, template, dest, modified_crates):
    ...
```

Reply:
done